### ISSUE TITLE
Hot reload for web app "pano" does not work with docker-compose and turbo

### ISSUE THREAD

**Issue by usirin** [kamp-us/monorepo#184]
## Current behavior

When using docker-compose and the monorepo tool "turbo" to run the "pano" web app located in the "apps/pano" directory, any changes made to the code are not reflected in the app unless the entire container is restarted. This is because the way "turbo" sets up the monorepo inside the docker container does not support hot reloading, which means that developers have to manually restart the container every time they make a change to the code.

## Expected behavior

It would be ideal if hot reloading could be supported by the combination of docker-compose and "turbo" for the "pano" web app, so that any changes made to the code would be reflected in the app without having to manually restart the container.

## Related issues

- #151 
- #124 
- #132 

## Additional information

This stopped after we switched to monorepo model using "turbo". I tried to follow https://turbo.build/repo/docs/handbook/deploying-with-docker for dev machines as well, but it might be just wrong because this optimizes layers for building for production. Dev containers might not need this special treatment actually.

cc @csoylu

**Comment by csoylu** [comment]
Tried to solve this issue today, a few problems.

## Ionotify events on docker WSL

Docker bind mounts, which we use in docker in order to mount source directories into the container, do not send Ionotify events when mounted from Windows filesystem.
The requst to resolve this issue, in docker repo, was  redirected to WSL in: https://github.com/docker/for-win/issues/12898 and https://github.com/docker/for-win/issues/12766

There is already an issue in WSL, in which we don't have any response from the maintainers if they will solve this or not: https://github.com/microsoft/WSL/issues/4739


### Solution in VSCode
We can copy the repo into the WSL filesystem, preferably into /home/user or /root. Then we can download the WSL extension from VSCODE, the extension asks if you want to connect through WSL and proceeds to create a vscode server in your WSL and connecting you to it.
Once you are connected through the server, when you try to open a file using ctrl+o you can choose files or folders in your WSL.
![image](https://user-images.githubusercontent.com/94180156/208055171-79cc4dfc-2dd9-4040-969d-5d59e543d89c.png)

Though you need to download all non-cosmetic extensions again.


## Fsevents on macOS
There are also some issues related to this for macs: https://github.com/docker/for-mac/issues/5755 and https://github.com/docker/for-mac/issues/4731

In the first issue, this was said;
> The best reference I could find for this behaviour is from [Mac OS X Internals: A Systems Approach, Section 11.8.2](https://books.google.co.uk/books?id=K8vUkpOXhN4C&pg=PA1416&lpg=PA1416&dq=fsevents+modify+when+file+is+closed&source=bl&ots=ONjjSZWpUD&sig=ACfU3U01Q6MhdwXffv9msYqeF1JzIaSimQ&hl=en&sa=X&ved=2ahUKEwj707K7xKvxAhWR8uAKHSq2CUgQ6AEwCXoECAoQAw#v=onepage&q=fsevents%20modify%20when%20file%20is%20closed&f=false) page 1416 "The Fsevents Mechanism" where it says:
> 
> FSE_CONTENT_MODIFIED: A file's content was modified -- specifically, a file that is being closed was written to while it was open.
> 
> Unfortunately I don't think there is anything we can do in Docker Desktop to work around this.

And in the second issue they said they were going to look into it but no update since.

## Conclusion
_Let's all use debian for developping 🥇_

I've also deleted the dockerfile, made some changes into the docker-compose.yml and added 3 scripts to monorepo package.json. Don't know if they are really necessary at this point.

### The dockerfile changes, for app:

```
    command: >
      bash -c "
      cd /app &&
      npm install -w apps/pano &&
      npm run pano:dev:docker
      "
    volumes:
      - app:/app/node_modules
      - /app/apps/pano/build
      - app_kampus_ui:/app/packages/kampus-ui/node_modules
      - /app/packages/kampus-ui/dist
      - ../../.:/app
    networks:
```
This mounts all the project, makes named volumes for node_modules and exclude some stuff so when something is written into it we dont receive it in the original project folder.
Node modules folders stay intact and are reused when container is launched, so it doesnt need to dowload each time. Though we can delete the volumes if we wish to start  from scratch.

### Added scripts in packages.json of monorepo
Basically runs pano and kampus-ui in development mode.
```
  "scripts": {
    "pano:dev:docker": "run-p pano:dev kampusui:dev",
    "pano:dev": "cd apps/pano && prisma generate && npm run docker",
    "kampusui:dev": "cd packages/kampus-ui && npm run dev"
  },
```

**Comment by csoylu** [comment]
I found it, so WSL is like what i said in previous comment.
For macos, i just tried it with @ickarakurt, we need to add port 8002 to docker compose and it works.
